docs: improve doc structure (#3488)
dnbexperience/eufemia: packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/fields.mdx

### Creating custom field components

The `useFieldProps` hook that is used in all existing field components is exported to make it possible [to create custom field components](/uilib/extensions/forms/create-component/) that have the same properties and follow the same flow as the standard components, without the need to recreate all the basic state handling features.

### Basic field usage

> Components in Eufemia Forms are very flexible. They adapt to the set of props they receive, and you can therefore choose which parts of the functionality they consist of that you use, and what you yourself want to handle in the application code.

Reviewer: ```suggestion
Components in Eufemia Forms are very flexible. They adapt to the set of props they receive, and you can therefore choose which parts of the functionality they should use, and what you yourself want to handle in the application code.
```

Here is an example that consists of a large degree of data handling in the application, even though the form itself still consists of both components from Eufemia and local special components (the blue boxes):

<div style={{ maxWidth: '1200px' }}>
  <InlineImg
    height="auto"
    width="auto"
    src={FormDiagram1}
    caption="Simple form example with Eufemia Forms fields"
  />
</div>

### Complete state management and local custom fields

> In this example, all state data, validation process and error handling are done by components from Eufemia Forms, while in a local field component created especially for this application, there is a combination of standardized field functionality taken from Eufemia Forms, and local special code to achieve the goal of maximum reuse without sacrificing flexibility:

Reviewer: ```suggestion
In this example, all state data, validation process and error handling are done by components from Eufemia Forms, inside a local field component created especially for this application. Here you find a combination of standardized field functionality taken from Eufemia Forms, and local specialized code to achieve the goal of maximum reuse without sacrificing flexibility:
```
